binderhub/static/js/index.js

```diff
@@ -29,121 +26,12 @@ import 'bootstrap/dist/css/bootstrap.min.css';
 import 'bootstrap/dist/css/bootstrap-theme.min.css';
 
 import '../index.css';
+import { setUpLog } from './src/log';
+import { updateUrls } from './src/urls';
+import { BASE_URL } from './src/constants';
+import { getBuildFormValues } from './src/form';
+import { updateRepoText } from './src/repo';
 
-const BASE_URL = $('#base-url').data().url;
-const BADGE_BASE_URL = $('#badge-base-url').data().url;
-let config_dict = {};
-
-function v2url(providerPrefix, repository, ref, path, pathType) {
-  // return a v2 url from a providerPrefix, repository, ref, and (file|url)path
-  if (repository.length === 0) {
-    // no repo, no url
-    return null;
-  }
-  let url;
-  if (BADGE_BASE_URL) {
-    url = BADGE_BASE_URL + 'v2/' + providerPrefix + '/' + repository + '/' + ref;
-  }
-  else {
-    url = window.location.origin + BASE_URL + 'v2/' + providerPrefix + '/' + repository + '/' + ref;
-  }
-  if (path && path.length > 0) {
-    // encode the path, it will be decoded in loadingMain
-    url = url + '?' + pathType + 'path=' + encodeURIComponent(path);
-  }
-  return url;
-}
-
-function loadConfig(callback) {
-  const req = new XMLHttpRequest();
-  req.onreadystatechange = function() {
-    if (req.readyState == 4 && req.status == 200)
-      callback(req.responseText)
-  };
-  req.open('GET', BASE_URL + "_config", true);
-  req.send(null);
-}
-
-function setLabels() {
-  const provider = $("#provider_prefix").val();
-  const text = config_dict[provider]["text"];
-  const tag_text = config_dict[provider]["tag_text"];
-  const ref_prop_disabled = config_dict[provider]["ref_prop_disabled"];
-  const label_prop_disabled = config_dict[provider]["label_prop_disabled"];
-  const placeholder = "HEAD";
-
-  $("#ref").attr('placeholder', placeholder).prop("disabled", ref_prop_disabled);
-  $("label[for=ref]").text(tag_text).prop("disabled", label_prop_disabled);
-  $("#repository").attr('placeholder', text);
-  $("label[for=repository]").text(text);
-}
-
-function updateRepoText() {
-  if (Object.keys(config_dict).length === 0){
-    loadConfig(function(res) {
-      config_dict = JSON.parse(res);
-      setLabels();
-    });
-  } else {
```

Comment:
This section previously looked like:

```js
  if (Object.keys(configDict).length === 0) {
    const configUrl = BASE_URL + "_config";
    fetch(configUrl).then(resp => {
      resp.json().then(data => {
        configDict = data
        setLabels();
      });
    })
  }
```

The loadConfig function now in repo.js is unchanged, looking like below with XMLHttpRequest being used still. I understand it as you've inlined the function into updateRepoText, and that updateRepoText was the only place where loadConfig was used, so I figure we should delete it.

```js
function loadConfig(callback) {
  const req = new XMLHttpRequest();
  req.onreadystatechange = function() {
    if (req.readyState == 4 && req.status == 200)
      callback(req.responseText)
  };
  req.open('GET', BASE_URL + "_config", true);
  req.send(null);
}
```

Reply:
Good catch. done